### ISSUE TITLE
Code to train on Microsoft Research Paraphrase Corpus(MSRP).

### ISSUE THREAD

**Issue by aironashish** [castorini/MP-CNN-Torch#1]
Hi, 
I was trying to replicate the results of your experiments on MSRP. But, I can't find the required code. Can you please suggest the required changes to handle the classification task. I guess, num_classes and loss function needs to be changed to 2 and hinge loss, respectively. 

Also, I was running trainSIC.lua on a dataset with 2 classes(and I made the required changes like changing num_classes = 2 and in predictCombination function val = torch.range(1,2,1)).But, the dev score results in NAN. But, if I run trainSIC without changing the Conv.lua and trainSIC.lua (dataset contains still 2 classes only). I am getting a dev score of .129.  Can you please explain the reason for this behavior?

**Comment by hohoCode** [comment]
Thanks for your interests. 

If you want to change the trainSIC.lua file for your own data set and for binary classification. I think it should be easy to do, besides what you have done, I think it is probably needed to pay attention to this part of codes:
https://github.com/hohoCode/textSimilarityConvNet/blob/master/Conv.lua#L89-L102

The training data labels should be set properly, please make sure the util/read_data.lua is able to handle your own dataset.

Also, in the prediction function, since your task is on binary classification, I do not think you still need the dot products as what we have done here, you may need to change the codes here as well:
https://github.com/hohoCode/textSimilarityConvNet/blob/master/Conv.lua#L142-L148

So overall I think you need to change Conv.lua slightly for your own data.

**Comment by aironashish** [comment]
Thank you for answering Part one. And second part is solved (changed few more lines).
